src/coreclr/jit/utils.h

```diff
@@ -1086,4 +1086,14 @@ bool CastFromFloatOverflows(float fromValue, var_types toType);
 bool CastFromDoubleOverflows(double fromValue, var_types toType);
 }
 
+template <typename T, typename U>
+T BitCast(const U& value)
```

Comment:
nit: +1 for this. Technically to be 100% UB free we may need extra trivially_copyable trait checks as in C++20 bit_cast.

Reply:
I agree with you but there is no `std::is_trivially_copyable` and `std::is_trivially_initializable` are not defined in the dotnet project so I can't use them :C. I believe that even that wouldn't be 100% UB free though, since standard doesn't define a behavior when the cast is done to the type, that doesn't cover a current value memory layout